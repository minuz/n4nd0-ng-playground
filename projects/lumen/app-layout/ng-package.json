{
  "$schema": "../../../node_modules/ng-packagr/ng-package.schema.json",
  "dest": "../../../dist/lumen/app-layout",
  "lib": {
    "entryFile": "src/public-api.ts"
  }
}