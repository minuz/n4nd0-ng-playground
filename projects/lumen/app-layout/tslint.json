{
  "extends": "../../../tslint.json",
  "rules": {
    "directive-selector": [
      true,
      "attribute",
      "lumen",
      "camelCase"
    ],
    "component-selector": [
      true,
      "element",
      "lumen",
      "kebab-case"
    ]
  }
}
