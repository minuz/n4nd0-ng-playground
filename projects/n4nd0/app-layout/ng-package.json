{
  "$schema": "../../../node_modules/ng-packagr/ng-package.schema.json",
  "dest": "../../../dist/n4nd0/app-layout",
  "lib": {
    "entryFile": "src/public-api.ts"
  }
}