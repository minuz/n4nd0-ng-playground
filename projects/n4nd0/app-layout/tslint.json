{
  "extends": "../../../tslint.json",
  "rules": {
    "directive-selector": [
      true,
      "attribute",
      "n4nd0",
      "camelCase"
    ],
    "component-selector": [
      true,
      "element",
      "n4nd0",
      "kebab-case"
    ]
  }
}
