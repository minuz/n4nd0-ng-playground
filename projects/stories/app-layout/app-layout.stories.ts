import { MatSidenavModule } from '@angular/material/sidenav';
import { MatToolbarModule } from '@angular/material/toolbar';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
import { AppLayoutComponent, AppLayoutService } from '@n4nd0/app-layout';
import { moduleMetadata, storiesOf } from '@storybook/angular';
import { withKnobs, boolean } from '@storybook/addon-knobs';

const stories = storiesOf('Layout', module);
stories.addDecorator(withKnobs);
stories.addDecorator(
  moduleMetadata({
    declarations: [AppLayoutComponent],
    imports: [BrowserAnimationsModule, MatSidenavModule, MatToolbarModule],
    providers: [AppLayoutService]
  })
);
stories.add('expanded', () => ({
  template: `
    <n4nd0-app-layout [toggleLeftSidenav]="toggleLeftSidenav" [toggleRightSidenav]="toggleRightSidenav">
      <div toolbar>Should show toolbar content</div>
      <div left-sidenav>Should display something on left sidenav</div>
      <div main-content>Should display something as main content</div>
      <div right-sidenav>Should display something as right sidenav</div>
    </n4nd0-app-layout>`,
  props: {
    toggleLeftSidenav: boolean('toggle left sidenav', true),
    toggleRightSidenav: boolean('toggle right sidenav', true)
  }
}));
