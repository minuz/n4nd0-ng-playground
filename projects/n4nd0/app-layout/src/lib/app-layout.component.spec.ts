import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { AppLayoutComponent } from './app-layout.component';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
import { MatSidenavModule } from '@angular/material/sidenav';
import { MatToolbarModule } from '@angular/material/toolbar';
import { AppLayoutService } from './app-layout.service';

describe('AppLayoutComponent', () => {
  let component: AppLayoutComponent;
  let fixture: ComponentFixture<AppLayoutComponent>;
  let service: AppLayoutService;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [AppLayoutComponent],
      imports: [BrowserAnimationsModule, MatSidenavModule, MatToolbarModule]
    })
      .compileComponents()
      .then(() => {
        fixture = TestBed.createComponent(AppLayoutComponent);
        component = fixture.componentInstance;
        fixture.detectChanges();
      });
  });

  test('should create', () => {
    expect(component).toBeTruthy();
  });

  describe('layout functionality', () => {
    beforeEach(async () => {
      service = component.layout;
    });

    test('Should be able to close left sidenav', () => {
      service.toggleLeftSidenav(false);
      fixture.detectChanges();
      expect(fixture.nativeElement).toMatchSnapshot();
    });
  });
});
