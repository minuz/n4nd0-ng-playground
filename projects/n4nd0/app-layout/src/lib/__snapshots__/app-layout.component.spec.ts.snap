// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppLayoutComponent layout functionality Should be able to close left sidenav 1`] = `
<div
  id="root1"
  ng-version="8.1.0"
>
  <mat-toolbar
    class="mat-toolbar mat-toolbar-single-row"
  />
  <mat-sidenav-container
    class="wrapper mat-drawer-container mat-sidenav-container"
  >
    
    <div
      aria-hidden="true"
      class="cdk-visually-hidden cdk-focus-trap-anchor"
    />
    <mat-sidenav
      class="mat-drawer mat-sidenav ng-tns-c5-0 ng-trigger ng-trigger-transform mat-drawer-side ng-star-inserted ng-animating"
      mode="side"
      ng-reflect-mode="side"
      ng-reflect-opened="false"
      ng-reflect-position="start"
      position="start"
      style=""
      tabindex="-1"
    >
      <div
        class="mat-drawer-inner-container"
      />
    </mat-sidenav>
    <div
      aria-hidden="true"
      class="cdk-visually-hidden cdk-focus-trap-anchor"
    />
    <div
      aria-hidden="true"
      class="cdk-visually-hidden cdk-focus-trap-anchor"
    />
    <mat-sidenav
      class="mat-drawer mat-sidenav ng-tns-c5-1 ng-trigger ng-trigger-transform mat-drawer-end mat-drawer-side ng-star-inserted"
      mode="side"
      ng-reflect-mode="side"
      ng-reflect-opened="false"
      ng-reflect-position="end"
      position="end"
      style="box-shadow:none;visibility:hidden;0:box-shadow;1:visibility;"
      tabindex="-1"
    >
      <div
        class="mat-drawer-inner-container"
      />
    </mat-sidenav>
    <div
      aria-hidden="true"
      class="cdk-visually-hidden cdk-focus-trap-anchor"
    />
    <mat-sidenav-content
      class="mat-drawer-content mat-sidenav-content"
    />
    
  </mat-sidenav-container>
</div>
`;
